src/RemoteInboxNotifications/GetRuleProcessor.php

```diff
@@ -59,6 +59,10 @@ public static function get_processor( $rule_type ) {
 				);
 			case 'data':
 				return new DataRuleProcessor();
+			case 'product_count':
+				return new ProductCountRuleProcessor(
+					new \WC_Product_Query( array( 'limit' => -1 ) )
```

Comment:
This is a super expensive query that could potentially cause issues on stores with lots of products. (We also don't need product data, just the count of products..)

Can we rely on the `total` property on the return object instead? https://github.com/woocommerce/woocommerce/blob/1d1413b4ee669cd37ee025c93972ce785a10a0b9/includes/data-stores/class-wc-product-data-store-cpt.php#L2002

Should we be filtering by publish status?

If we can rely on the `total` property, we should be able to reduce the `limit` to `1` instead of ALL products (`-1`).

Aside: I was going to suggest we use `wp_count_posts()` as the [`WC_Install` class does](https://github.com/woocommerce/woocommerce/blob/4b3977868b789b83800f33e06c4ade1da5738c0e/includes/class-wc-install.php#L341), but that won't work if a store has replaced the data store for products.

Reply:
Nice! I've done this in 8db0229f55e7e0cedad824e10d5ebb0a39207f40